### ISSUE TITLE
feature request: a summary of the executions

### ISSUE THREAD

**Issue by scoopex** [puppetlabs/bolt#272]
It would be nice to have a summary of all executions performed in a bolt run.

This would be useful for the management of problems while exection and to gather information from hosts. The summary should be reusable as a list of hosts.

Example:
If you gather information about swapspace useage of your systems by the following execution it is hard if you use bolt on many hundreds of hosts:

```
bolt command run "swapon -s|grep partition" -n `hget all`
```

A result like that would be nice:
```
$ bolt command run "swapon -s|grep partition" -n `hget all`
....
Ran on 216 nodes in 13.60 seconds
Successful nodes (exitcode 0): foo-ca-001,foo-cdh01,foo-cdh02,foo-cdh03,foo-ci01,foo-computing-001,foo-drop02,foo-git01-001, ....
Failed nodes (exitcode not 0): foo-vm190,foo-vm191,foo-vm192,foo-vm194,foo-vm195,foo-vm196,foo-vm197,foo-vm198...

$ bolt command run "swapoff -a" -n oo-vm190,foo-vm191,foo-vm192,foo-vm194,foo-vm195,foo-vm196,foo-vm197,foo-vm198...
```

**Comment by MikaelSmith** [comment]
I'm not sure what that will look like as an output format.

It's probably easier to do with a plan to aggregate the results right now (which provides tools for accessing structured results) or by processing the JSON output (via `--format json`).

**Comment by MikaelSmith** [comment]
Oh, just a summary of successful and failed nodes at the end? That wouldn't be too bad.

**Comment by scoopex** [comment]
Yes, especially if the ouput-list is re-usable a new "--node" argument.

**Comment by MikaelSmith** [comment]
Presumably this would apply to command/script/task, but not plan. Because plans may not have a definitive list of passed/failed nodes.

**Comment by MikaelSmith** [comment]
[BOLT-345](https://tickets.puppetlabs.com/browse/BOLT-345) opened to cover this.

**Comment by MikaelSmith** [comment]
This has been fixed in Bolt 0.17.0.

**Comment by scoopex** [comment]
Thats cool, thanks :-)
